### PR TITLE
feat: added kafka - scenario - 4

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

Added scenario - 4 changes

#### Screenshots

https://github.com/user-attachments/assets/c6af450a-f60c-42bb-b636-e5b3bb59aaa4

#### Affected Areas and Manually Tested Areas

----

> [!IMPORTANT]
> Add 'dropRate' view to Kafka messaging queues with new API and UI updates.
> 
>   - **Behavior**:
>     - Added 'dropRate' view to `MQDetailPage.tsx`, replacing 'consumerLatency'.
>     - Updated `MessagingQueuesDetails` to exclude 'dropRate' from certain views.
>     - Introduced `getKafkaSpanEval` in `getKafkaSpanEval.tsx` for 'dropRate' API calls.
>   - **API**:
>     - New API function `getKafkaSpanEval` for evaluating Kafka spans.
>     - Modified `getConsumerLagDetails.ts` to include optional `evalTime`.
>   - **UI Components**:
>     - Updated `MessagingQueueOverview.tsx` to handle 'dropRate' view.
>     - Adjusted `MessagingQueues.tsx` to reflect 'dropRate' in UI.
>   - **Utilities**:
>     - Added 'dropRate' to `MessagingQueuesViewType` in `MessagingQueuesUtils.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for ba898890a4cc8bde1fca7eb7402dd6b283af0c90. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: frontend/src/pages/MessagingQueues/MQDetails/MessagingQueueOverview.tsx [hunk 1]
```diff
@@ -78,6 +82,10 @@ function MessagingQueueOverview({
 					? 'producer'
 					: 'consumer'
 				: undefined,
+		evalTime:
+			selectedView === MessagingQueuesViewType.dropRate.value
+				? 2363404
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetails/MessagingQueueOverview.tsx:87]: ignore this, it has been taken care in followup PR

File: frontend/src/pages/MessagingQueues/MQDetails/MQTables/getKafkaSpanEval.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+		return {
+			statusCode: 200,
+			error: null,
+			message: response.data.status,
+			payload: response.data.data,
+		};
+	} catch (error) {
+		return ErrorResponseHandler((error as AxiosError) || SOMETHING_WENT_WRONG);
+	}
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetails/MQTables/getKafkaSpanEval.tsx:33]: added error handling for all APIs and tables - like this - 
image

[the remark this answers is not in the document]
Comment (ahmadshaheer) [reply frontend/src/pages/MessagingQueues/MQDetails/MQTables/getKafkaSpanEval.tsx]: :+1:  :pray: :raised_hands:

**Review by rahulkeswani101 — APPROVED** [review]

**Review by ahmadshaheer — APPROVED** [review]
Thanks for making the changes, @SagarRajput-7 :pray: :slightly_smiling_face:.

assuming that you have tested the error states.

**Comment by SagarRajput-7** [comment]
> Thanks for making the changes, @SagarRajput-7 🙏 🙂.
> 
> assuming that you have tested the error states.

Yes

**Review by YounixM — APPROVED** [review]

**Pull request merged**